plugins/module_utils/network/sonic/config/dhcp_relay/dhcp_relay.py

```diff
@@ -0,0 +1,490 @@
+
+            ipv4 = command.get('ipv4')
+            ipv6 = command.get('ipv6')
+            if intf_name and not ipv4 and not ipv6:
+                if have_ipv4:
+                    requests.append({'path': self.dhcp_relay_intf_config_path['server_addresses_all'].format(intf_name=intf_name), 'method': DELETE})
+                if have_ipv6:
+                    requests.append({'path': self.dhcpv6_relay_intf_config_path['server_addresses_all'].format(intf_name=intf_name), 'method': DELETE})
+            else:
+                if ipv4 and have_ipv4:
+                    requests.extend(self.get_delete_specific_dhcp_relay_param_requests(command, have_obj))
+                if ipv6 and have_ipv6:
+                    requests.extend(self.get_delete_specific_dhcpv6_relay_param_requests(command, have_obj))
+
+        return requests
+
+    def get_delete_specific_dhcp_relay_param_requests(self, command, config):
+        """Get requests to delete specific DHCP relay configurations
+        based on the command specified for the interface
+        """
+        requests = []
+
+        name = command['name']
+        ipv4 = command.get('ipv4')
+        have_ipv4 = config.get('ipv4')
+        if not ipv4 or not have_ipv4:
+            return requests
+
+        server_addresses = self.get_server_addresses(ipv4.get('server_addresses'))
+        have_server_addresses = self.get_server_addresses(have_ipv4.get('server_addresses'))
+
+        # Delete all DHCP relay config for an interface, if only
+        # an address with no value is specified.
+        if (ipv4.get('server_addresses') and len(ipv4.get('server_addresses'))
+                and not server_addresses):
+            requests.append({'path': self.dhcp_relay_intf_config_path['server_addresses_all'].format(intf_name=name), 'method': DELETE})
+            return requests
+
+        del_server_addresses = have_server_addresses.intersection(server_addresses)
+        if del_server_addresses:
+            # Delete all DHCP relay config for an interface if all
+            # server addresses are to be deleted
+            if len(del_server_addresses) == len(have_server_addresses):
+                requests.append({'path': self.dhcp_relay_intf_config_path['server_addresses_all'].format(intf_name=name), 'method': DELETE})
+                return requests
+
+            for addr in del_server_addresses:
+                url = self.dhcp_relay_intf_config_path['server_address'].format(intf_name=name, server_address=addr)
+                requests.append({'path': url, 'method': DELETE})
+
+        # Specifying appropriate order for deletion to succeed
+        if ipv4.get('link_select') is not None and have_ipv4.get('link_select'):
+            url = self.dhcp_relay_intf_config_path['link_select'].format(intf_name=name)
+            requests.append({'path': url, 'method': DELETE})
+
+        if ipv4.get('source_interface') and have_ipv4.get('source_interface'):
```

Comment:
Please add verification that the source addresses match before enqueuing this deletion request.

Reply:
Done